tools/src/gtwapi-manifests/pin.go

```diff
@@ -0,0 +1,140 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build pin
+// +build pin
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"sync"
+
+	"helm.sh/helm/v3/pkg/kube"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/cli-runtime/pkg/resource"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	addToScheme sync.Once
+	nopLogger   = func(_ string, _ ...interface{}) {}
+)
+
+func newClient() *kube.Client {
+	addToScheme.Do(func() {
+		if err := apiextv1.AddToScheme(scheme.Scheme); err != nil {
+			// This should never happen.
+			panic(err)
+		}
+		if err := apiextv1beta1.AddToScheme(scheme.Scheme); err != nil {
+			panic(err)
+		}
+	})
+
+	testFactory := cmdtesting.NewTestFactory()
+
+	c := &kube.Client{
+		Factory: testFactory.WithNamespace("default"),
+		Log:     nopLogger,
+	}
+
+	return c
+}
+
+func main() {
+	outputDir := env("OUTPUT_DIR", "charts/gateway-helm")
+	gtwAPIVer := env("GATEWAY_API_VERSION", "v0.7.1")
```

Comment:
auto read current gwapi version?

Reply:
it's passed from Makefile

https://github.com/envoyproxy/gateway/blob/76f66f4a37899af358fe294b41f67e6b9dd6a027/tools/make/kube.mk#L38